src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -480,20 +523,9 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
                 // If we didn't get a nuget service, there's nothing we can do in terms of querying the solution for
                 // nuget info.
                 if (nugetService == null)
-                    return;
-
-                // Figure out the entire set of projects to process.
-                using var _ = PooledHashSet<ProjectId>.GetInstance(out var projectsToProcess);
+                    return default;
 
-                var solution = _workspace.CurrentSolution;
-                AddProjectsToProcess(workQueue, solution, projectsToProcess);
-
-                // And Process them one at a time.
-                foreach (var projectId in projectsToProcess)
-                {
-                    cancellationToken.ThrowIfCancellationRequested();
-                    await ProcessProjectChangeAsync(nugetService, solution, projectId, cancellationToken).ConfigureAwait(false);
-                }
+                return await doWorkAsync(nugetService).ConfigureAwait(false);
```

Comment:
would it be reasonable here to make the function just take in a cancellation token?  To avoid any chance that the cancellation token in dowork is different from the one passed into PerformNuGetProjectServiceWorkAsync

Reply:
sure, that woudl be reasonable :)